packages/manager/apps/web/client/app/hosting/cdn/shared/cors/routing.js

```diff
@@ -0,0 +1,27 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('app.hosting.dashboard.cdn.shared.cors', {
+    url: '',
+    layout: 'modal',
+    params: {
+      cors: null,
+    },
+    views: {
+      modal: {
+        component: 'hostingCdnSharedSettingsCors',
+      },
+    },
+    resolve: {
+      cors: /* @ngInject */ ($transition$) => $transition$.params().cors,
+      goBack: /* @ngInject */ ($state) => () => $state.go('^'),
+
+      onValidate: /* @ngInject */ (cors, goBack) => (origins) => {
+        const allOrigins = Array.isArray(origins) && origins.length === 0;
+        Object.assign(
+          cors.config,
+          allOrigins ? { origins: '*' } : { origins: origins.join(',') },
+        );
```

Comment:
Why enrich object if not used elsewhere ?

Reply:
The object `cors` is used by the parent state to validate later the settings